### ISSUE TITLE
Incorrect generated xref link

### ISSUE THREAD

**Issue by robbiemc** [speced/respec#4787]
## Description of problem

The IsolatedContext IDL attribute resolves correctly in bikeshed and [respec.org/xref](https://respec.org/xref/?term=IsolatedContext&types=extended-attribute) after https://github.com/w3c/browser-specs/pull/1467, but it doesn't resolve to the correct URL when generating a spec with respec.

respec.org/xref generates its link by [resolving](https://github.com/speced/respec-web-services/blob/main/static/xref/script.js#L150) the search uri against the spec's url, which does the right thing.

When respec looks up the xref info from respec.org/xref/search it gets:
```
{
    "shortname": "isolated-contexts",
    "spec": "isolated-contexts",
    "type": "extended-attribute",
    "normative": true,
    "uri": "#IsolatedContext"
}
```
It then [resolves](https://github.com/speced/respec/blob/main/src/core/xref.js#L402) `uri` against `https://partial`, and pulls the path and hash out of that, which in this case ends up setting the path to '/'. Later, when generating the anchor's href, the specification URL's path gets [overridden](https://github.com/speced/respec/blob/main/src/core/data-cite.js#L59) because of the '/'.

**URL to affected spec or repo:**

https://github.com/WICG/direct-sockets/pull/71

**What happened (e.g., it crashed)?:**

`{{IsolatedContext}}` resolved to the wrong link.

**Optional, steps to reproduce:**

1. Add `xref: ["isolated-contexts"],` to respecConfig
2. Include `{{IsolatedContext}}` in a spec
3. Check the href that IsolatedContext resolves to

**Comment by robbiemc** [comment]
Thanks for the quick fix! Any objections to releasing a new version with the fix?

**Comment by sidvishnoi** [comment]
I'll release tomorrow.

**Comment by sidvishnoi** [comment]
Released [`v35.1.2`](https://github.com/speced/respec/releases/tag/v35.1.2)
